components/mbedtls/Kconfig

```diff
@@ -1,5 +1,27 @@
 menu "mbedTLS"
 
+config MBEDTLS_PLATFORM_MEMORY
+   bool "Enable mbedTLS memory allocation layer."
+   default n
+   help
+      By default mbed TLS uses the system-provided calloc() and free().
+      This allows different allocators (self-implemented or provided) to be
+      provided to the platform abstraction layer.
+
+      Enabling MBEDTLS_PLATFORM_MEMORY without the
```

Comment:
Most of this description doesn't make sense in the ESP-IDF context, where the user isn't editing the mbedTLS config header and can't easily define macros. Suggest something like:

> If this option is disabled, mbed TLS uses the default system calloc() and free() functions.
> 
> If this option is enabled, the mbed TLS config macro MBEDTLS_PLATFORM_MEMORY will be defined. The function mbedtls_platform_set_calloc_free() must be called at runtime to provide custom calloc() and free() function pointers for use by mbedTLS.
>
> This option allows fine-grained control over how mbedTLS allocates heap memory.